src/screens/settings/blockchain-portfolio/blockchain-portfolio.tsx

```diff
@@ -0,0 +1,129 @@
+};
+
+const mapStateToProps = (state: IReduxState) => {
+    return {
+        blockchains: Object.keys(state.app.blockchains)
+            .map(key => ({ key, value: state.app.blockchains[key] }))
+            .sort((a, b) => a.value.order - b.value.order)
+    };
+};
+
+const navigationOptions = ({ theme }: any) => ({
+    title: translate('Settings.blockchainPortfolio'),
+    headerStyle: {
+        backgroundColor: themes[theme].colors.header
+    }
+});
+
+export class BlockchainPortfolioComponent extends React.Component<
+    INavigationProps & IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
+> {
+    public static navigationOptions = navigationOptions;
+
+    public renderBlockchain(
+        item: { key: Blockchain; value: IBlockchainsOptions },
+        move: any,
+        moveEnd: any
+    ) {
+        const { styles, theme } = this.props;
+
+        return (
+            <View style={styles.rowContainer}>
+                <View style={styles.infoContainer}>
+                    <Icon name="money-wallet-1" size={ICON_SIZE} style={styles.blockchainIcon} />
+                    <Text style={styles.blockchainName}>{capitalizeFirstLetter(item.key)}</Text>
+                </View>
+                <TouchableOpacity
+                    style={styles.iconContainer}
+                    onPress={() => this.props.toogleBlockchainActive(item.key)}
+                >
+                    <Icon
+                        size={18}
+                        name={item.value.active ? 'check-2-thicked' : 'check-2'}
+                        style={[
+                            {
+                                color: item.value.active
+                                    ? theme.colors.accent
+                                    : theme.colors.textSecondary
+                            }
+                        ]}
+                    />
+                </TouchableOpacity>
+                <TouchableOpacity
+                    style={styles.iconContainer}
+                    onLongPress={move}
+                    onPressOut={moveEnd}
+                >
+                    <Icon size={18} name="navigation-menu" style={styles.menuIcon} />
+                </TouchableOpacity>
+            </View>
+        );
+    }
+
+    public render() {
+        const { styles } = this.props;
+
+        return (
+            <View style={styles.container}>
+                <DraggableFlatList
+                    data={this.props.blockchains}
+                    renderItem={({ item, move, moveEnd }) =>
+                        this.renderBlockchain(item, move, moveEnd)
+                    }
+                    keyExtractor={(item: { key: Blockchain; value: IBlockchainsOptions }) =>
+                        `${item.value.order}`
+                    }
+                    scrollPercent={5}
+                    onMoveEnd={({ data }) => {
+                        this.props.updateBlockchainOrder(
+                            Object.assign(
```

Comment:
i this was the only place where reduce was making sense :))
It's fine like this also.

Reply:
Pff true :)) I understood now